### PR TITLE
Use `ci-jobs` Utility For `ci.yml` Matrix

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This pull request is a follow-up to https://github.com/woocommerce/woocommerce/pull/43345. In that PR I added a new `ci-jobs` command to the `@woocommerce/monorepo-utils` utility. That command replaces the current CI job matrix script with something that is more robust, easier to maintain, and well-tested. In continuing with this line of work I have now made the following changes:

* **`@woocommerce/monorepo-utils` Changes**
  * Reworked the config a little bit to better conform to `ci.yml` requirements.
  * Fixed a couple of bugs that weren't caught previously.
  * Switch build to `webpack` and included the bundled build in the repository. This means that we don't need to run `pnpm install` in order to use the utility anymore. I did this so that we could use it in CI without adding a minute to the runtime for the `ci.yml` workflow.
  * Corrected `bin` packaging and made sure everything works as expected.
* **`ci.yml` Changes**
  * The `ci-jobs` command returns linting and testing as separate jobs. This is deliberate as it makes failures easier to follow when looking at the jobs list.
  * Updated the workflow to take advantage of the new command.
* **`package.json` Changes**
  * The new `config.ci` objects have been filled out according to the needs of each project. These should include information about how to run each command as well as the changes that should trigger each command. 
    * **Please review these thoroughly since people won't notice if a change doesn't trigger a workflow that it should!**

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Testing this pull request requires creating a PR targeting this branch and making various changes to files to trigger tests. I will do my best to include some examples of this after creation. 

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

This is a CI-only change.
 
</details>

### CODE REVIEW

**Comment by jonathansadowski** [comment]
I'll be taking a closer look at this shortly. In the meantime @ObliviousHarmony, just to verify, the current expected checks not finishing is expected for now, as after this is merged, we'd update all of those separate expected checks and replace them with a single "Evaluate Project Job Statuses" expected check, is that correct?

**Comment by ObliviousHarmony** [comment]
Yes @jonathansadowski, it's because I renamed the job.

**Comment by rrennick** [comment]
> just to verify, the current expected checks not finishing is expected for now, as after this is merged, we'd update all of those separate expected checks and replace them with a single "Evaluate Project Job Statuses" expected check, is that correct?

This is the list of jobs picked up by this PR with `pnpm utils ci-jobs trunk`. My trunk might be slightly out of date.

```
-  @woocommerce/monorepo-utils - lint
-  @woocommerce/data - lint
-  @woocommerce/product-editor - lint
-  @woocommerce/admin-library - lint
-  @woocommerce/block-library - lint
-  @woocommerce/plugin-woocommerce - lint:php:changes:branch trunk
Test Jobs
-  @woocommerce/monorepo-utils - JavaScript
-  @woocommerce/data - JavaScript
-  @woocommerce/components - JavaScript
-  @woocommerce/product-editor - JavaScript
-  @woocommerce/admin-library - JavaScript
-  @woocommerce/block-library - JavaScript
-  @woocommerce/plugin-woocommerce - PHP
-  @woocommerce/plugin-woocommerce - PHP 8.0
-  @woocommerce/plugin-woocommerce - PHP WP: nightly
-  @woocommerce/plugin-woocommerce - PHP WP: latest - 1
-  @woocommerce/plugin-woocommerce - PHP WP: latest - 2
```

**Review by rrennick — COMMENTED** [review]
@jonathansadowski This looks good to me but I'll leave it for you to approve.

**Comment by rrennick** [comment]
When I tried to switch way from this branch I got the following error:

```
error: Your local changes to the following files would be overwritten by checkout:
	tools/monorepo-utils/fonts/Standard.flf
```

Should this file be added to the PR or to `.gitignore`?

File: packages/js/api-core-tests/package.json [hunk 1]
```diff
@@ -49,5 +49,18 @@
...
+	},
+	"config": {
+		"ci": {
+			"lint": {
+				"command": "lint",
+				"changes": [
+					"data/**/*.js",
+					"endpoints/**/*.js",
+					"tests/**/*.js",
+					"utils/**/*.js"
```
Comment (jonathansadowski) [inline packages/js/api-core-tests/package.json]: Currently these directories only contain `.js` files, but the lint command itself contains the args `--ext=js,ts,tsx`, should we be more inclusive here?

**Comment by ObliviousHarmony** [comment]
> Should this file be added to the PR or to .gitignore?

No @rrennick, this file is necessary in order for the `@woocommerce/monorepo-utils` utility to work without being built. We've got the `.gitattributes` to automatically set the line ending, however, it seems that `git` gives some messaging when it needs to do this conversion depending on your configuration. I've added a step to the build for the package that automatically sets the line endings to `\n` and that should solve any issues relating to this.

File: packages/js/api-core-tests/package.json (hunk 1 above)
Comment (ObliviousHarmony) [reply packages/js/api-core-tests/package.json]: Good catch, I must have decided to check the `--ext` option later on in the process of converting these.

**Review by jonathansadowski — APPROVED** [review]
I've tested this with several changes that trigger both lint and test jobs. I've also reviewed all of the `package.json` changes, and they look correct to me. I didn't both trigger a test job and lint job for each project, but I've tested out the various scenarios, and this looks like it's good to go to me.

Nice work!

**Pull request merged**